sqlair.go

```diff
@@ -103,6 +103,21 @@ func (q *Query) Run() error {
 	return nil
 }
 
+// RunQuery is shorthand for preparing a statement, building a query, then doing Query.Run.
+func (db *DB) RunQuery(ctx context.Context, query string, inputArgs ...any) error {
```

Comment:
This issue with `Iter`, `All` and `One` is that they have output expressions as well and `Prepare` needs all the types mentioned in the query, including these.
If we have `QueryString` which returns a `Query` then we don't see the output types until they call `.All` ect.

We could have them passed to `QueryString` next to the input arguments but this would be a little messy, you would pass input types only once but output types twice.

If we did have separate replicates of each method that would work, but I agree that this isnt ideal either. 

Since `QueryString` doesn't return an error it could be feasible to somehow delay the `Prepare` until we have all the types (i.e. they call `All`/`Iter`/`One`) but I'm not sure what the best mechanism to do this might be.